src/wallet/delete_wallet_flow.c

```diff
@@ -0,0 +1,166 @@
+#include "tasks.h"
+#include "ui_core_confirm.h"
+#include "ui_screens.h"
+#include "ui_state_machine.h"
+#include "wallet.h"
+#include "wallet_list.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+extern char *ALPHA_NUMERIC;
+
+extern Wallet wallet;
+extern Wallet_credential_data wallet_credential_data;
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+static bool get_pin_input() {
+  if (!WALLET_IS_PIN_SET(wallet.wallet_info)) {
+    return true;
+  }
+
+  memzero(&wallet_credential_data, sizeof(wallet_credential_data));
+
+  input_text_init(ALPHA_NUMERIC, ui_text_enter_pin, 4, DATA_TYPE_PIN, 8);
+  if (0 == get_state_on_input_scr(0, 1, 2)) {
+    sha256_Raw((uint8_t *)flow_level.screen_input.input_text,
+               strnlen(flow_level.screen_input.input_text,
+                       sizeof(flow_level.screen_input.input_text)),
+               wallet_credential_data.password_single_hash);
+    sha256_Raw(wallet_credential_data.password_single_hash,
+               SHA256_DIGEST_LENGTH,
+               wallet.password_double_hash);
+
+    memzero(flow_level.screen_input.input_text,
+            sizeof(flow_level.screen_input.input_text));
+    return true;
+  }
+
+  return false;
+}
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+void delete_wallet_flow(const uint8_t *wallet_id) {
+  ASSERT(NULL != wallet_id);
+
+  if (!core_scroll_page(
+          NULL, ui_text_need_all_x1cards_to_delete_wallet_entirely, NULL)) {
+    return;
+  }
+
+  memzero(&wallet, sizeof(wallet));
+  memzero(&wallet_shamir_data, sizeof(wallet_shamir_data));
+  memzero(&wallet_credential_data, sizeof(wallet_credential_data));
```

Comment:
Use common function implemented in [view seed flow](https://github.com/Cypherock/x1_wallet_firmware/pull/339)

Reply:
Added, was waiting for merges